File: iteration1/src/Models/Course.java
package iteration1.src.Models;

import iteration1.src.Resources.CourseType;

import java.util.ArrayList;

// This class will store courses.
public class Course {
    private String courseName;
    private CourseCode courseCode;
    private int credit;
    private ArrayList<CourseCode> preRequisiteCourses;
    private ArrayList<CourseSession> courseSessions;
    private int requiredCredits;
    private Semester courseSemester;
    private CourseType courseType;

    // Course will be created according to the data in the JSON file.
    public Course(String courseName, CourseCode courseCode, int credit, ArrayList<CourseCode> preRequisiteCourses, ArrayList<CourseSession> courseSessions, int requiredCredits, Semester courseSemester, CourseType courseType) {
        this.courseName = courseName;
        this.courseCode = courseCode;
        this.credit = credit;
        this.preRequisiteCourses = preRequisiteCourses;
        this.courseSessions = courseSessions;
        this.requiredCredits = requiredCredits;
        this.courseSemester = courseSemester;
        this.courseType = courseType;
    }

    // These are general getter methods, there is nothing different.
    public String getCourseName() {
        return courseName;
    }

    public CourseCode getCourseCode() {
        return courseCode;
    }

    public int getCredit() {
        return credit;
    }

    public ArrayList<CourseCode> getPreRequisiteCourses() {
        return preRequisiteCourses;
    }

    public ArrayList<CourseSession> getCourseSessions() {
        return courseSessions;
    }

    public int getRequiredCredits() {
        return requiredCredits;
    }

    public Semester getCourseSemester() {
        return courseSemester;
    }

    public CourseType getCourseType() {
        return courseType;
    }

    // toString returns course name, course code and credit.
    @Override
    public String toString() {
        return "Course Name: " + courseName + "---Course Code: " + courseCode + "---Credit: " + credit;
    }
}

File: python_project/src/Models/Course.py
class Course:

    def __init__(self, courseName, courseCode, credit, requiredCredits, courseSemester, courseType, preRequisisteCourses=None, courseSessions=None):
        self.courseName = courseName
        self.courseCode = courseCode
        self.credit = credit
        self.requiredCredits = requiredCredits
        self.courseSemester = courseSemester
        self.courseType = courseType

        if preRequisisteCourses is None:
            self.preRequisisteCourses = []
        else:
            self.preRequisisteCourses = preRequisisteCourses

        if courseSessions is None:
            self.courseSessions = []
        else:
            self.courseSessions = courseSessions

    def __eq__(self, other):
        return self.courseName == other.courseName and self.courseCode == other.courseCode and self.credit == other.credit and self.requiredCredits == other.requiredCredits and self.courseSemester == other.courseSemester and self.courseType == other.courseType and self.preRequisisteCourses == other.preRequisisteCourses and self.courseSessions == other.courseSessions

    def __str__(self):
        return f"{self.courseName} --Course Code: {self.courseCode} --Credit: {self.credit}"
